apps/tx-builder/src/components/TransactionDetails.tsx

```diff
@@ -0,0 +1,183 @@
+
+  const isCustomHexDataTx = !!hexEncodedData;
+  const isContractInteractionTx = !!contractMethod && !isCustomHexDataTx;
+
+  const isTokenTransferTx = !isCustomHexDataTx && !isContractInteractionTx;
+
+  return (
+    <Wrapper>
+      <StyledTxTitle size="md">
+        {isTokenTransferTx ? `Transfer ${weiToEther(value)} ${nativeCurrencySymbol} to:` : 'Interact with:'}
+      </StyledTxTitle>
+
+      <EthHashInfo shortName={networkPrefix || ''} hash={to} showAvatar showCopyBtn shouldShowShortName />
+
+      <TxSummaryContainer>
+        {/* to address */}
+        <Text size="xl" color="secondaryHover">
+          to (address)
+        </Text>
+        <EthHashInfo shortName={networkPrefix || ''} hash={to} shortenHash={4} showCopyBtn shouldShowShortName />
+
+        {/* value */}
+        <Text size="xl" color="secondaryHover">
+          value:
+        </Text>
+        <TxValueLabel>{`${weiToEther(value)} ${nativeCurrencySymbol}`}</TxValueLabel>
+
+        {/* data */}
+        <Text size="xl" color="secondaryHover">
+          data:
+        </Text>
+        <TxValueLabel>{data}</TxValueLabel>
+
+        {isContractInteractionTx && (
+          <>
+            {/* method */}
+            <Text size="xl" color="secondaryHover">
+              method:
+            </Text>
+            <StyledTxValueLabel size="xl">{contractMethod.name}</StyledTxValueLabel>
+
+            {/* method params */}
+            {contractMethod.inputs.map(({ name, type }, index) => {
+              const methodName = name || index;
+              const methodLabel = `${methodName} (${type})`;
+              return (
+                <React.Fragment key={`${methodName}`}>
+                  {/* param name */}
+                  <StyledMethodNameLabel size="xl" color="secondaryHover" tooltip={methodLabel}>
+                    {methodLabel}
+                  </StyledMethodNameLabel>
+                  {/* param value */}
+                  <TxValueLabel>{contractFieldsValues?.[methodName]}</TxValueLabel>
+                </React.Fragment>
+              );
+            })}
+          </>
+        )}
+      </TxSummaryContainer>
+    </Wrapper>
+  );
+};
+
+export default TransactionDetails;
+
+const Wrapper = styled.article`
+  flex-grow: 1;
+  padding: 0 16px;
+  user-select: text;
+`;
+
+const TxSummaryContainer = styled.div`
+  display: grid;
+  grid-template-columns: minmax(100px, 2fr) minmax(100px, 5fr);
+  gap: 4px;
+
+  margin-top: 16px;
+`;
+
+const StyledTxTitle = styled(Title)`
+  font-size: 16px;
+  margin: 8px 0;
+  font-weight: bold;
+  line-height: initial;
+`;
+
+const StyledMethodNameLabel = styled(Text)`
+  padding-left: 4px;
+
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+
+const LINE_HEIGHT = 22;
+const MAX_HEIGHT = 2 * LINE_HEIGHT; // 2 lines as max height
+
+const TxValueLabel = ({ children }: { children: React.ReactNode }) => {
+  const [showMore, setShowMore] = useState(false);
```

Comment:
I created this small component to implement de `Show more` feature, its a bit tricky and I tried to be as clean as I could